src/badguy/root.cpp

```diff
@@ -16,79 +16,290 @@
 
 #include "badguy/root.hpp"
 
-#include "sprite/sprite.hpp"
-#include "sprite/sprite_manager.hpp"
+#include "audio/sound_manager.hpp"
+#include "math/random.hpp"
+#include "object/sprite_particle.hpp"
+#include "supertux/sector.hpp"
 
-static const float SPEED_GROW = 256;
-static const float SPEED_SHRINK = 128;
-static const float HATCH_TIME = 0.75;
+static const float HATCH_TIME = 0.7f;
+static const float APPEAR_TIME = 0.5f;
+static const float RETREAT_TIME = 1.f;
 
-Root::Root(const Vector& pos, Flip flip) :
-  BadGuy(pos, "images/creatures/ghosttree/root.sprite", LAYER_TILES-1),
-  mystate(STATE_APPEARING),
-  base_sprite(SpriteManager::current()->create("images/creatures/ghosttree/root-base.sprite")),
-  offset_y(0),
-  hatch_timer()
+Root::Root(const ReaderMapping& reader) :
+  BadGuy(reader, "images/creatures/mole/corrupted/root.sprite" , LAYER_TILES-10)
 {
-  base_sprite->set_action("appearing", 1);
-  base_sprite->set_animation_loops(1); // TODO: Necessary because set_action ignores loops for default actions.
+  construct();
+}
+
+Root::Root(const Vector& pos, Direction dir, const std::string& sprite,
+           float delay, bool play_sound) :
+  BadGuy(pos, dir, sprite, LAYER_TILES-10)
+{
+  construct(delay, play_sound);
+}
+
+void
+Root::construct(float delay, bool play_sound)
+{
+  m_base_surface = nullptr;
+  m_state = STATE_HATCHING;
+  m_delay = (delay == -1 ? HATCH_TIME : delay);
+  m_maxheight = 0.f;
+  m_play_sound = play_sound;
```

Comment:
Member variables should be initialized in constructor initializer list.

Reply:
I didn't want to copy code so i put it here.

Reply:
They should, it's a convention followed everywhere.